### DISCUSSION TITLE
Cant get cache to work with Openplotter / galadrielmap_sk/

### DISCUSSION THREAD

**Discussion by mrbc42** [VladimirKalachikhin/Galadriel-cache#5] [General]
Hi there, thank you for this program, I am trying to get this to work and unsuccessful.  Could you provide any pointers, this really seems like a great solution for me and I am sure I am missing something simple.
thanks,

So far:
- Installed open plotter on RPI4-4g
- Installed apache2 and php7
- added galadrielcache to html root
- can browse galadrielcache dir structure
- Installed /galadrielmap_sk/ plugin
- installed Signalk charts plugin
- Added http://10.10.10.1/tileproxy/tiles.php?z={z}&x={x}&y={y}&r=OpenNauticalChart to Online providers

I can see the map name in the list of maps but no map data will download.

Error in apache2 logs is 
[Tue Jun 18 23:27:51.386788 2024] [php7:notice] [pid 792] [client 10.10.10.63:43898] PHP Notice:  Undefined variable: content_encoding in /var/www/html/tileproxy/tiles.php on line 134, referer: http://10.10.10.1:3000/
PHP Notice:  Only variables should be passed by reference in /var/www/html/tileproxy/tilefromsource.php on line 266
PHP Notice:  Undefined index: checkonly in /var/www/html/tileproxy/tilefromsource.php on line 375
[Tue Jun 18 23:27:54.184863 2024] [php7:notice] [pid 793] [client 10.10.10.63:43900] PHP Notice:  Undefined variable: ContentType in /var/www/html/tileproxy/tiles.php on line 134, referer: http://10.10.10.1:3000/
[Tue Jun 18 23:27:54.184941 2024] [php7:notice] [pid 793] [client 10.10.10.63:43900] PHP Notice:  Undefined variable: content_encoding in /var/www/html/tileproxy/tiles.php on line 134, referer: http://10.10.10.1:3000/
[Tue Jun 18 23:30:42.566658 2024] [php7:notice] [pid 794] [client 10.10.10.63:54824] PHP Notice:  Undefined variable: mapSourcesName in /var/www/html/tileproxy/mapsources/navionics.php on line 34, referer: http://10.10.10.1:3000/
[Tue Jun 18 23:30:42.566795 2024] [php7:notice] [pid 794] [client 10.10.10.63:54824] PHP Notice:  Undefined variable: functionGetTileFile in /var/www/html/tileproxy/tiles.php on line 43, referer: http://10.10.10.1:3000/
[Tue Jun 18 23:30:42.566883 2024] [php7:notice] [pid 794] [client 10.10.10.63:54824] PHP Notice:  Undefined variable: functionPrepareTileImg in /var/www/html/tileproxy/tiles.php on line 112, referer: http://10.10.10.1:3000/
[Tue Jun 18 23:30:42.585188 2024] [php7:notice] [pid 795] [client 10.10.10.63:54840] PHP Notice:  Undefined variable: functionGetTileFile in /var/www/html/tileproxy/tiles.php on line 43, referer: http://10.10.10.1:3000/
[Tue Jun 18 23:30:42.585354 2024] [php7:notice] [pid 795] [client 10.10.10.63:54840] PHP Notice:  Undefined variable: functionPrepareTileImg in /var/www/html/tileproxy/tiles.php on line 112, referer: http://10.10.10.1:3000/
PHP Warning:  require_once(fNavionics.php): failed to open stream: No such file or directory in /var/www/html/tileproxy/tilefromsource.php(147) : eval()'d code on line 1
PHP Fatal error:  require_once(): Failed opening required 'fNavionics.php' (include_path='.:/usr/share/php') in /var/www/html/tileproxy/tilefromsource.php(147) : eval()'d code on line 1
[Tue Jun 18 23:30:42.640336 2024] [php7:notice] [pid 794] [client 10.10.10.63:54824] PHP Notice:  Undefined variable: ContentType in /var/www/html/tileproxy/tiles.php on line 134, referer: http://10.10.10.1:3000/
[Tue Jun 18 23:30:42.640402 2024] [php7:notice] [pid 794] [client 10.10.10.63:54824] PHP Notice:  Undefined variable: content_encoding in /var/www/html/tileproxy/tiles.php on line 134, referer: http://10.10.10.1:3000/
PHP Notice:  Only variables should be passed by reference in /var/www/html/tileproxy/tilefromsource.php on line 266
PHP Notice:  Undefined index: checkonly in /var/www/html/tileproxy/tilefromsource.php on line 375
[Tue Jun 18 23:30:45.188984 2024] [php7:notice] [pid 795] [client 10.10.10.63:54840] PHP Notice:  Undefined variable: ContentType in /var/www/html/tileproxy/tiles.php on line 134, referer: http://10.10.10.1:3000/
[Tue Jun 18 23:30:45.189061 2024] [php7:notice] [pid 795] [client 10.10.10.63:54840] PHP Notice:  Undefined variable: content_encoding in /var/www/html/tileproxy/tiles.php on line 134, referer: http://10.10.10.1:3000/

**Comment by VladimirKalachikhin** [comment]
Something strange...
A see: ` require_once(fNavionics.php): failed to open stream: `, but OpenNauticalChart don't use fNavionics.php.

**Comment by mrbc42** [comment]
I deleted the original dir and copied all the tilemap files from the docker container to my RPI and manually set all the permissions.  Now I get tiles but they are all the same(see image).  I get this error also.

Any ideas? 


PHP Notice:  Only variables should be passed by reference in /var/www/html/tileproxy/tilefromsource.php on line 266
PHP Notice:  Undefined index: checkonly in /var/www/html/tileproxy/tilefromsource.php on line 375
 tilefromsource.php getTile: Saved 334 bytes to ./tiles/OpenNauticalChart/5/13/17.png from http://t1.openseamap.org/seamark/5/13/17.png
[Wed Jun 19 22:55:49.938075 2024] [php7:notice] [pid 2761] [client 10.10.10.63:53048] PHP Notice:  Undefined variable: ContentType in /var/www/html/tileproxy/tiles.php on line 127, referer: http://10.10.10.1:3000/
[Wed Jun 19 22:55:49.938162 2024] [php7:notice] [pid 2761] [client 10.10.10.63:53048] PHP Notice:  Undefined variable: content_encoding in /var/www/html/tileproxy/tiles.php on line 127, referer: http://10.10.10.1:3000/
PHP Notice:  Only variables should be passed by reference in /var/www/html/tileproxy/tilefromsource.php on line 266
PHP Notice:  Undefined index: checkonly in /var/www/html/tileproxy/tilefromsource.php on line 375
 tilefromsource.php getTile: Saved 334 bytes to ./tiles/OpenNauticalChart/5/10/13.png from http://t1.openseamap.org/seamark/5/10/13.png
[Wed Jun 19 22:55:49.950335 2024] [php7:notice] [pid 2686] [client 10.10.10.63:53038] PHP Notice:  Undefined variable: ContentType in /var/www/html/tileproxy/tiles.php on line 127, referer: http://10.10.10.1:3000/
[Wed Jun 19 22:55:49.950421 2024] [php7:notice] [pid 2686] [client 10.10.10.63:53038] PHP Notice:  Undefined variable: content_encoding in /var/www/html/tileproxy/tiles.php on line 127, referer: http://10.10.10.1:3000/




drwxrwxrwx 5 www-data www-data  4096 Jun 19 22:15 tileproxy

root@openplotter:/var/www/html/tileproxy# ls -al
total 268
drwxrwxrwx 5 www-data www-data  4096 Jun 19 22:15 .
drwxr-xr-x 3 root     root      4096 Jun 19 22:15 ..
-rw-rw-r-- 1 root     root      1823 Jun 19 22:15 checkSources.php
-rw-rw-rw- 1 www-data www-data  3548 Jun 19 22:15 clearCache.php
-rw-rw-r-- 1 root     root     10998 Jun 19 22:15 fcommon.php
-rw-rw-rw- 1 www-data www-data  2764 Jun 19 22:15 fNavionics.php
-rw-rw-rw- 1 www-data www-data   498 Jun 19 22:15 info.php
-rw-r--r-- 1 root     root     19133 Jun 19 22:15 LICENSE.md
-rw-r--r-- 1 root     root     43101 Jun 19 22:15 LICENSE.ru-RU.md
drwxrwxrwx 4 www-data www-data  4096 Jun 19 22:44 loaderJobs
-rw-rw-rw- 1 www-data www-data 13857 Jun 19 22:15 loader.php
-rw-rw-r-- 1 root     root     14869 Jun 19 22:15 loaderSched.php
drwxrwxrwx 2 www-data www-data  4096 Jun 19 22:15 mapsources
-rw-rw-r-- 1 root     root      4730 Jun 19 22:15 mapsourcesVariablesList.php
-rw-rw-r-- 1 root     root      1941 Jun 19 22:15 MBTilesClient.php
-rw-rw-r-- 1 root     root     11377 Jun 19 22:15 MBTilesServer.php
-rw-rw-r-- 1 www-data www-data  5052 Jun 19 22:15 params.php
-rw-r--r-- 1 root     root      9528 Jun 19 22:15 README.md
-rw-r--r-- 1 root     root     19366 Jun 19 22:15 README.ru-RU.md
-rw-rw-r-- 1 root     root     28962 Jun 19 22:15 tilefromsource.php
drwxrwxrwx 5 www-data www-data  4096 Jun 19 22:42 tiles
-rw-rw-r-- 1 root     root      4441 Jun 19 22:15 tilesCOVER.php
-rw-rw-r-- 1 root     root     16589 Jun 19 22:15 tiles.php
root@openplotter:/var/www/html/tileproxy# 

![tile](https://github.com/VladimirKalachikhin/Galadriel-cache/assets/6986403/66d5a199-99b0-4857-8af9-46d4e02f537c)

**Comment by VladimirKalachikhin** [comment]
But in the picture, it's map Navionics...

Pls, get the tileproxy files from this archive:
https://github.com/VladimirKalachikhin/Galadriel-map/raw/master/emergencykit/GaladrielMap-2.9_GaladrielCache-2_gpsdPROXY-0_netAIS-1_NoVectorTiles_.zip
Remove all tiles from `tileproxy/tiles/` directory.

It is better to set the error_reporting variable in php.ini to  Default Value.

**Comment by mrbc42** [comment]
hi, I Removed tileproxy dir completly and replaced with files from archive above.  Set the php.ini to default value for logging.

Same issue with tiles displayed.  I would be grateful for any other ideas.
```
[Thu Jun 20 04:17:36.237900 2024] [php7:notice] [pid 3193] [client 10.10.10.63:34636] PHP Notice:  Undefined variable: mapSourcesName in /var/www/html/tileproxy/mapsources/navionics_layer.php on line 75, referer: http://10.10.10.1:3000/
[Thu Jun 20 04:17:36.238101 2024] [php7:notice] [pid 3193] [client 10.10.10.63:34636] PHP Notice:  Undefined variable: functionGetTileFile in /var/www/html/tileproxy/tiles.php on line 43, referer: http://10.10.10.1:3000/
[Thu Jun 20 04:17:36.238247 2024] [php7:notice] [pid 3193] [client 10.10.10.63:34636] PHP Notice:  Undefined variable: functionPrepareTileImg in /var/www/html/tileproxy/tiles.php on line 112, referer: http://10.10.10.1:3000/
PHP Warning:  file_put_contents(./tiles/navionics_layer/navtoken): failed to open stream: No such file or directory in /var/www/html/tileproxy/tilefromsource.php(147) : eval()'d code on line 45
PHP Notice:  Only variables should be passed by reference in /var/www/html/tileproxy/tilefromsource.php on line 266
PHP Notice:  Undefined index: checkonly in /var/www/html/tileproxy/tilefromsource.php on line 375
 tilefromsource.php getTile: Saved 19116 bytes to ./tiles/navionics_layer/12/2374/1161.png from http://backend.navionics.com/tile/12/2374/1161?LAYERS=config_1_10.00_0&TRANSPARENT=TRUE&UGC=FALSE&navtoken=eyJrZXkiOiJOYXZpb25pY3NfaW50ZXJuYWxwdXJwb3NlXzAwMDAxIiwia2V5RG9tYWluIjoid2ViYXBwLm5hdmlvbmljcy5jb20iLCJyZWZlcmVyIjoid2ViYXBwLm5hdmlvbmljcy5jb20iLCJyYW5kb20iOjE3MTg4NTM0NTgyMjV9
[Thu Jun 20 04:17:37.803461 2024] [php7:notice] [pid 3193] [client 10.10.10.63:34636] PHP Notice:  Undefined variable: ContentType in /var/www/html/tileproxy/tiles.php on line 134, referer: http://10.10.10.1:3000/
[Thu Jun 20 04:17:37.803578 2024] [php7:notice] [pid 3193] [client 10.10.10.63:34636] PHP Notice:  Undefined variable: content_encoding in /var/www/html/tileproxy/tiles.php on line 134, referer: http://10.10.10.1:3000/

```

```
root@openplotter:/var/www/html/tileproxy# ls -al
total 268
drwxr-xr-x 5 www-data www-data  4096 Jun 20 04:15 .
drwxr-xr-x 3 root     root      4096 Jun 20 03:24 ..
-rw-r--r-- 1 www-data www-data  1823 Jun 20 03:24 checkSources.php
-rw-r--r-- 1 www-data www-data  3548 Jun 20 03:24 clearCache.php
-rw-r--r-- 1 www-data www-data 11210 Jun 20 03:24 fcommon.php
-rw-r--r-- 1 www-data www-data  2764 Jun 20 03:24 fNavionics.php
-rw-r--r-- 1 www-data www-data   498 Jun 20 03:24 info.php
-rw-r--r-- 1 www-data www-data 19133 Jun 20 03:24 LICENSE.md
-rw-r--r-- 1 www-data www-data 43101 Jun 20 03:24 LICENSE.ru-RU.md
drwxrwxr-x 4 www-data www-data  4096 Jun 20 03:24 loaderJobs
-rw-r--r-- 1 www-data www-data 13857 Jun 20 03:24 loader.php
-rw-r--r-- 1 www-data www-data 14869 Jun 20 03:24 loaderSched.php
drwxr-xr-x 2 www-data www-data  4096 Jun 20 03:24 mapsources
-rw-r--r-- 1 www-data www-data  4730 Jun 20 03:24 mapsourcesVariablesList.php
-rw-r--r-- 1 www-data www-data  1941 Jun 20 03:24 MBTilesClient.php
-rw-r--r-- 1 www-data www-data 11377 Jun 20 03:24 MBTilesServer.php
-rw-r--r-- 1 www-data www-data  5147 Jun 20 04:12 params.php
-rw-r--r-- 1 www-data www-data  9491 Jun 20 03:24 README.md
-rw-r--r-- 1 www-data www-data 19331 Jun 20 03:24 README.ru-RU.md
-rw-r--r-- 1 www-data www-data 29114 Jun 20 03:24 tilefromsource.php
drwxrwxrwx 5 www-data www-data  4096 Jun 20 04:17 tiles
-rw-r--r-- 1 www-data www-data  4456 Jun 20 03:24 tilesCOVER.php
-rw-r--r-- 1 www-data www-data 16718 Jun 20 03:24 tiles.php

```
signalK charts provider 
http://10.10.10.1/tileproxy/tiles.php?z=12&x=2374&y=1161&r=navionics_layer



![nav_layer](https://github.com/VladimirKalachikhin/Galadriel-cache/assets/6986403/c4cfaac3-470e-46d1-9af3-8087c52a6d18)


```
root@openplotter:/var/www/html/tileproxy# cat params.php 
<?php
/* Version 2.7.4
Пути и параметры
Options and paths 
*/
// Общие параметры Common
// Пути paths
//	Хранилище тайлов, каталог в файловой системе. Обычно -- ссылка на специальный раздел.
$tileCacheDir = 'tiles'; 	// tile storage directory, in filesystem. Link is good idea.
//	Каталог описаний источников карт, в файловой системе
$mapSourcesDir = 'mapsources'; 	// map sources directory, in filesystem.
// Параметры options
//	Умолчальный срок хранения тайла в кеше, прежде чем будет предпринята попытка его обновления, сек. 86400 сек. == 1 ден. Если 0 -- никогда не переписывать тайл. В параметрах источника эта величина может быть изменена.
$ttl = 86400*365*3; //default cache timeout in seconds.  86400 sec == 1 day. After this, tile trying to reload from source. If 0 - never trying to reload.
// Срок, через который будет предпринята попытка вновь получить тайл, который получить по какой-то причине не удалось. Если 0 -- не пытаться.
$noTileReTry = 86400; //in seconds.  86400 sec == 1 day. If no tile -- retry after this time. If 0 - never trying to reload.
// 	Умолчальное расширение файла тайла
$ext = 'png'; 	// default tile image type/extension
// 	Умолчальный минимальный масштаб. Тайлы меньшего масштаба скачиваться не будут, если это не указано в описании источника карты
$minZoom = 0; 	// default min zoom
// 	Умолчальный максимальный масштаб. Тайлы масштаба крупнее скачиваться не будут
$maxZoom = 19; 	// default max zoom
// 	Число попыток скачать тайл, прежде чем он будет считаться отсутствующим
$maxTry = 3; 	// number of tryes to download tile from source
// 	Пауза между попытками скачать тайл, сек.
$tryTimeout = 3; 	// pause between try to download tile from source, sec
// 	Время ожидания ответа источника тайла, сек.
$getTimeout = 10; 	// timeout tile source response, sec
// 	Пауза между запросами, если обнаружено отсутствие связи
$noInternetTimeout = 20; 	// no try the source this time if no internet connection found, sec

// Глобальный прокси -- для всех источников
//$globalProxy = 'tcp://127.0.0.1:8118'; 	// Global Proxy. May be tor via Polipo, for example. If not defined - not used.

// Глобальный список мусорных тайлов. Тайлы с такой суммой crc32 заменяются пустыми, но будет предпринята попытка их получить снова через $noTileReTry

// crc32 of junk tiles '00000000' zero length file and '0940c426' empty png are not trash!
$globalTrash = array(
'f821973a',	// белый тайл. Борьба с русскими?
'a5da4659',	// белый тайл
'af8e80fa'	// прозрачный тайл.
);
 
// Загрузчик Tile loader.
// Пути paths
// 	Путь к каталогу заданий планировщика, в файловой системе
$jobsDir = 'loaderJobs'; 	// loader jobs directory, in filesystem. Check the user rights to this directory. Need to read/write for loaderSched.php
// 	Путь к каталогу заданий загрузчика, в файловой системе 
$jobsInWorkDir = "$jobsDir/inWork"; 	// current jobs directory.  Check the user rights to this directory. Need to read/write for loaderSched.php and loader.php
// Параметры options
// 	Количество одновременно запускаемых загрузчиков
$maxLoaderRuns = 10; 	// simultaneously working loader tasks. Set at least 2 to avoid blocking download by bad source.
// 	Загрузчик скачивает тайлы до масштаба включительно
$loaderMaxZoom = 16; 	// loader download tiles to this zoom only, not to map or default $maxZoom
// 	Предварительная загрузка тайлов большего масштаба запускается при просмотре тайла указанного масштаба
$aheadLoadStartZoom = 14; // start of the ahead loading from this zoom 

// Определение покрытия Tiles cover detection
$tileCacheServerPath = '/tileproxy'; 	// web path to GaladrielCache

// Ситемные параметры System
// Вызов php из командной строки
//$phpCLIexec = 'php-cli'; 	// php-cli executed name on your OS
$phpCLIexec = 'php'; 	// php-cli executed name on your OS
?>

root@openplotter:/etc/apache2/sites-enabled# cat 000-default.conf 
<VirtualHost *:80>
	# The ServerName directive sets the request scheme, hostname and port that
	# the server uses to identify itself. This is used when creating
	# redirection URLs. In the context of virtual hosts, the ServerName
	# specifies what hostname must appear in the request's Host: header to
	# match this virtual host. For the default virtual host (this file) this
	# value is not decisive as it is used as a last resort host regardless.
	# However, you must set it for any further virtual host explicitly.
	#ServerName www.example.com

	ServerAdmin webmaster@localhost
	DocumentRoot /var/www/html

	# Available loglevels: trace8, ..., trace1, debug, info, notice, warn,
	# error, crit, alert, emerg.
	# It is also possible to configure the loglevel for particular
	# modules, e.g.
	#LogLevel info ssl:warn

	ErrorLog ${APACHE_LOG_DIR}/error.log
	CustomLog ${APACHE_LOG_DIR}/access.log combined

	# For most configuration files from conf-available/, which are
	# enabled or disabled at a global level, it is possible to
	# include a line for only one particular virtual host. For example the
	# following line enables the CGI configuration for this host only
	# after it has been globally disabled with "a2disconf".
	#Include conf-available/serve-cgi-bin.conf
</VirtualHost>

**Comment by VladimirKalachikhin** [comment]
In signalK charts provider, the URL must be:
http://10.10.10.1/tileproxy/tiles.php?z={z}&x={x}&y={y}&r=navionics_layer
not
http://10.10.10.1/tileproxy/tiles.php?z=12&x=2374&y=1161&r=navionics_layer

**Comment by mrbc42** [comment]
Thank you Vladimir, its working. Such a great program you have created.
